addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/handler/BaseKeypadHandler.java

```diff
@@ -195,12 +197,49 @@ private void asyncInitialize() {
 
             configureChannels();
 
-            updateStatus(ThingStatus.ONLINE);
+            initDeviceState();
 
             logger.debug("Async init thread finishing for keypad handler {}", integrationId);
         }
     }
 
+    public void initDeviceState() {
+        synchronized (asyncInitLock) {
+            Bridge bridge = getBridge();
+            if (bridge == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+            } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+                if (ledList.isEmpty()) {
+                    // Device with no LEDs has nothing to query. Assume it is online if bridge is online.
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    // Query LED states. Method handleUpdate() will set thing status to online when response arrives.
+                    updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, "Awaiting initial response");
+                    // To reduce query volume, query only 1st LED and LEDs with linked channels.
+                    for (KeypadComponent component : ledList) {
+                        if (component.id() == ledList.get(0).id() || isLinked(channelFromComponent(component.id()))) {
+                            queryDevice(component.id(), ACTION_LED_STATE);
+                        }
+                    }
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
```

Comment:
Did you think about moving this method to the `LutronHandler` instead of implement it in each handler?

Reply:
Yes, that's what I had intended to do originally.  But there were some complications with putting it there, so I'm afraid I took the lazy way out.  I've now done it properly. :-)